tests/integration/features/cleanup_int_test.go

```diff
@@ -69,18 +70,99 @@ var _ = Describe("feature cleanup", func() {
+			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace)).
 				WithTimeout(fixtures.Timeout).
 				WithPolling(fixtures.Interval).
 				Should(WithTransform(errors.IsNotFound, BeTrue()))
 		})
 
 	})
 
+	Context("cleaning up conditionally enabled features", Ordered, func() {
+
+		const (
+			featureName = "enabled-conditionally"
+			secretName  = "test-secret"
+		)
+
+		var (
+			dsci            *dsciv1.DSCInitialization
+			namespace       string
+			featuresHandler *feature.FeaturesHandler
+		)
+
+		BeforeAll(func() {
+			namespace = envtestutil.AppendRandomNameTo("test-conditional-cleanup")
+			dsci = fixtures.NewDSCInitialization(namespace)
+		})
+
+		It("should create feature, apply resource and create feature tracker", func() {
+			// given
+			err := fixtures.CreateOrUpdateNamespace(envTestClient, fixtures.NewNamespace("conditional-ns"))
+			Expect(err).To(Not(HaveOccurred()))
+
+			featuresHandler = feature.ClusterFeaturesHandler(dsci, func(handler *feature.FeaturesHandler) error {
+				conditionalCreationErr := feature.CreateFeature(featureName).
+					For(handler).
+					UsingConfig(envTest.Config).
+					PreConditions(
+						feature.CreateNamespaceIfNotExists(namespace),
+					).
+					EnabledWhen(namespaceExists).
+					WithResources(fixtures.CreateSecret(secretName, namespace)).
+					Load()
+
+				Expect(conditionalCreationErr).ToNot(HaveOccurred())
+
+				return nil
+			})
+
+			// when
+			Expect(featuresHandler.Apply()).Should(Succeed())
+
+			// then
+			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace)).
+				WithTimeout(fixtures.Timeout).
+				WithPolling(fixtures.Interval).
+				Should(Succeed())
+		})
+
+		It("should clean up resources when the condition is no longer met", func() {
+			// given
+			err := envTestClient.Delete(context.Background(), fixtures.NewNamespace("conditional-ns"))
+			Expect(err).To(Not(HaveOccurred()))
+
+			// Mimic reconcile by re-loading the feature handler
+			featuresHandler = feature.ClusterFeaturesHandler(dsci, func(handler *feature.FeaturesHandler) error {
+				conditionalCreationErr := feature.CreateFeature(featureName).
+					For(handler).
+					UsingConfig(envTest.Config).
+					PreConditions(
+						feature.CreateNamespaceIfNotExists(namespace),
+					).
+					EnabledWhen(namespaceExists).
+					WithResources(fixtures.CreateSecret(secretName, namespace)).
+					Load()
+
+				Expect(conditionalCreationErr).ToNot(HaveOccurred())
+
+				return nil
+			})
+
+			Expect(featuresHandler.Apply()).Should(Succeed())
+
+			// then
+			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace)).
+				WithTimeout(fixtures.Timeout).
+				WithPolling(fixtures.Interval).
+				Should(WithTransform(errors.IsNotFound, BeTrue()))
+		})
+	})
 })
 
-func createdSecretHasOwnerReferenceToOwningFeature(namespace, secretName string) func() error {
+func createdSecretHasOwnerReferenceToOwningFeature(namespace string) func() error {
```

Comment:
In `createdSecretHasOwnerReferenceToOwningFeature` shouldn't we ensure we look up the FT matching the feature, not assuming there's a FT matching any Feature in the cluster that was related to "test-secret"? In other words I would think this function also expects a Feature to be passed by only reading the signature. Otherwise we assume a bit too much internally.